descriptor/qsaratomic/src/main/java/org/openscience/cdk/qsar/descriptors/atomic/IPAtomicHOSEDescriptor.java

```diff
@@ -331,20 +331,19 @@ public double extractIP(IAtomContainer container, IAtom atom) {
      */
     private static List<String> extractInfo(String str) {
 
-        StringBuffer idEdited = new StringBuffer();
-        StringBuffer valEdited = new StringBuffer();
+        StringBuilder idEdited  = new StringBuilder();
+        StringBuilder valEdited = new StringBuilder();
 
         int strlen = str.length();
 
         boolean foundSpace = false;
-        int countSpace = 0;
         boolean foundDigit = false;
         for (int i = 0; i < strlen; i++) {
-            if (!foundDigit) if (Character.isLetter(str.charAt(i))) foundDigit = true;
```

Comment:
Not changed in your patch, so no worries, but I really don't understand what this code is supposed to do... not very useful variable naming :(

Reply:
Well that's the part that's failing.... :-) it's basically trying to split the input into <hose_code><space><id>

Reply:
It was failing but fixed now, or was failing and still is? This line you did not change, just reformat, right? Or am I blind. My point was `foundDigit` suggests it's looking for digits... but isLetter() is not doing that, or am I missing something?

Reply:
No not changed yet, currently failing.